### PR TITLE
Bytes::with_capacity

### PR DESCRIPTION
**Pull request by stepancheg**

### CODE REVIEW

File: src/bytes.rs [hunk 1]
```diff
@@ -1554,7 +1582,7 @@ impl<'a> Extend<&'a u8> for BytesMut {
 
 impl Inner {
     #[inline]
-    fn empty() -> Inner {
+    fn _empty() -> Inner {
```
Comment (carllerche) [inline src/bytes.rs]: Why was this renamed? It seems that it wasn't in use?

Comment (stepancheg) [inline src/bytes.rs]: It was in use, and it is not used after this commit.

Comment (carllerche) [inline src/bytes.rs]: Would you mind deleting the fn?

Comment (stepancheg) [inline src/bytes.rs]: Done.
